File: Module_2.3/CASA/src/main/java/com/example/deposit/service/AccountService.java
package com.example.deposit.service;

import com.example.deposit.dto.AccountResponse;
import com.example.deposit.dto.CasaAccountRequest;
import com.example.deposit.dto.TermDepositRequest;
import com.example.deposit.entity.DepositAccount;
import com.example.deposit.entity.TermDeposit;

import java.math.BigDecimal;
import java.util.List;

public interface AccountService {

    AccountResponse createCasaAccount(CasaAccountRequest request);

    AccountResponse createTermDepositAccount(TermDepositRequest request);

    DepositAccount getDepositAccountById(Long accountId);

    TermDeposit getTermDepositByAccountId(Long accountId);

    TermDeposit getTermDepositByTdId(Long tdId);
    List<TermDeposit> getTdsMaturingToday();

    List<DepositAccount> getAccountsByCustomerId(Long customerId);
    List<DepositAccount> getAllActiveCasaAccounts();

    void updateBalance(Long accountId, BigDecimal balance);

    void updateTermDepositStatus(Long tdId, String status);
}

File: frontend/src/services/accountService.js
import api from './api'

export const createCasaAccount = (data) =>
  api.post('/v1/accounts/casa', data).then(r => r.data)

export const createTermDeposit = (data) =>
  api.post('/v1/accounts/term-deposit', data).then(r => r.data)

export const getAccountsByCustomerId = (customerId) =>
  api.get(`/v1/accounts/customer/${customerId}`).then(r => r.data)

export const getAccountById = (id) =>
  api.get(`/v1/accounts/${id}`).then(r => r.data)

export const getTermDepositByAccountId = (accountId) =>
  api.get(`/v1/accounts/term-deposit/${accountId}`).then(r => r.data)

export const getTermDepositByTdId = (tdId) =>
  api.get(`/v1/accounts/term-deposit/by-tdid/${tdId}`).then(r => r.data)
